packages/hardhat-chai-matchers/src/internal/reverted/revertedWith.ts

```diff
@@ -1,45 +1,64 @@
+import { AssertionError } from "chai";
 import { decodeReturnData, getReturnDataFromError } from "./utils";
 
+const buildAssert =
```

Comment:
Maybe de-duplicate this by extracting it to some shared module?

Reply:
Yes! I will move this to another module and add a long-ass comment explaining why we are doing this monstrosity.